portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -1532,7 +1532,7 @@ private void _checkFileEntry(
 				FileEntry.class.getName());
 
 		fileEntryModelResourcePermission.check(
-			permissionChecker, fileEntry.getFileEntryId(), ActionKeys.VIEW);
+			permissionChecker, fileEntry.getFileEntryId(), ActionKeys.DOWNLOAD);
```

Comment:
Can you elaborate a little bit? I don't understand the use case.

Reply:
The use case is this:
 I want to use an image for the home page (view permission for guest) but I remove the download permission (for guest or all) .
On this case the image should be shown on the page, but the image is not because guest user has not download permission (but the image has view permission)
You are always running `_checkFileEntry` method, that has the check of the download permission, and if the image doesn't have this permission, the image is not shown

Reply:
This is the expected behavior. Removing download permissions won't allow to view the content (as that requires a download). Note that adaptive media generated images will still be visible; this last point is a requirement so that previews can be displayed.

Reply:
this is so confusing...